napari/_tests/test_view_layers.py

```diff
@@ -164,3 +164,40 @@ def test_kwargs_passed(monkeypatch):
         call(title='my viewer'),
         call().open(path='some/path', name='img name', scale=(1, 2, 3)),
     ]
+
+
+# plugin_manager fixture is added to prevent errors due to installed plugins
+def test_imshow(qtbot, napari_plugin_manager):
+    shape = (10, 15)
+    ndim = len(shape)
+    np.random.seed(0)
+    data = np.random.random(shape)
+    viewer, layer = napari.imshow(data, show=False)
+    view = viewer.window._qt_viewer
+    check_viewer_functioning(viewer, view, data, ndim)
+    viewer.close()
+
+
+# plugin_manager fixture is added to prevent errors due to installed plugins
+def test_imshow_multichannel(qtbot, napari_plugin_manager):
+    """Test adding image."""
+    np.random.seed(0)
+    data = np.random.random((15, 10, 5))
+    viewer, layers = napari.imshow(data, channel_axis=-1, show=False)
+    assert len(layers) == data.shape[-1]
+    for i in range(data.shape[-1]):
+        assert np.all(layers[i].data == data.take(i, axis=-1))
+    viewer.close()
+
+
+# plugin_manager fixture is added to prevent errors due to installed plugins
+def test_imshow_with_viewer(qtbot, napari_plugin_manager, make_napari_viewer):
+    shape = (10, 15)
+    ndim = len(shape)
+    np.random.seed(0)
+    data = np.random.random(shape)
+    viewer = make_napari_viewer()
+    viewer, layer = napari.imshow(data, viewer=viewer, show=False)
```

Comment:
I think we should test that the returned viewer is the same that was passed in, and that the layer data is the same as as the data that was given:

```suggestion
    viewer = make_napari_viewer()
    viewer2, layer = napari.imshow(data, viewer=viewer, show=False)
    assert viewer is viewer2
    np.testing.assert_array_equal(data, layer.data)
```

I use np testing in the suggestion in case we ever decide to wrap the NumPy arrays e.g. with an evented NumPy array.